src/Spout/Writer/Style/BorderBuilder.php

```diff
@@ -0,0 +1,75 @@
+<?php
+
+namespace Box\Spout\Writer\Style;
+
+/**
+ * Class BorderBuilder
+ */
+class BorderBuilder
+{
+    /**
+     * @var Border
+     */
+    protected $border;
+
+    public function __construct()
+    {
+        $this->border = new Border();
+    }
+
+    /**
+     * @param string $style Border style @see BorderPart::allowedStyles
+     * @param string $color Border A RGB color code
+     * @param string $width Border width @see BorderPart::allowedWidths
+     * @return self
+     */
+    public function setBorderTop($style = Border::STYLE_SOLID, $color = Color::BLACK, $width = Border::WIDTH_THICK)
```

Comment:
Well, we can push as is but changing the order of the params is a breaking change. So if we do make the change after the current change was released, we'll need to bump to the next major version just for that. This is something I'd like to avoid.
I'm fine pushing as is for now to avoid this commit from being too big and taking too long to be merged. But I won't release a new version until we get this ordering/default setters sorted out.